src/core/Akka/Serialization/NewtonSoftJsonSerializer.cs

```diff
@@ -138,51 +172,75 @@ public NewtonSoftJsonSerializer(ExtendedActorSystem system, Config config)
             : this(system, NewtonSoftJsonSerializerSettings.Create(config))
         {
         }
-
-
+        
         public NewtonSoftJsonSerializer(ExtendedActorSystem system, NewtonSoftJsonSerializerSettings settings)
             : base(system)
         {
-            Settings = new JsonSerializerSettings
+            if (settings.UsePooledStringBuilder)
+            {
+                _sbPool = new DefaultObjectPool<StringBuilder>(
+                    new StringBuilderPooledObjectPolicy()
+                    {
+                        InitialCapacity = settings.StringBuilderMinSize,
+                        MaximumRetainedCapacity =
+                            settings.StringBuilderMaxSize
+                    });
+            }
+            Settings = CreateInternalSettings(system, settings,this);
+            var settingsNoFormat = CreateInternalSettings(system, settings,this);
+            settingsNoFormat.Formatting = Formatting.None;
+            _serializer = JsonSerializer.Create(Settings);
+            _formattingNoneSerializer = JsonSerializer.Create(settingsNoFormat);
```

Comment:
so, the idea here is to make sure that even if the user wants formatting for -their- serializer, we have our own with no formatting, so that we avoid making a new serializer every time.